pkg/apiserver/watch.go

```diff
@@ -196,7 +197,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				// client disconnect.
 				return
 			}
-			flusher.Flush()
+			if len(ch) == 0 {
```

Comment:
Flush will happen under the covers for us - if we had enough items that the
cluster would never be full, we would never have to flush because Go would
be flushing for us.

On Mon, May 23, 2016 at 1:11 PM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/apiserver/watch.go
> <https://github.com/kubernetes/kubernetes/pull/26085#discussion_r64252864>
> :
>
> > @@ -196,7 +197,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
> >  				// client disconnect.
> >  				return
> >  			}
> > -			flusher.Flush()
> > +			if len(ch) == 0 {
>
> I'm not convinced about this change. If the chrun in the cluster is big
> enough, ch can potentially never be empty.
>
> I think we should flush no matter what at least 1 per X milliseconds.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26085/files/c4bec1585fe9c31303c928c11ac45e380d2eaf77#r64252864>
>